### ISSUE TITLE
QR code scanner - android.hardware.Camera is deprecated

### ISSUE THREAD

**Issue by bbusenius** [hahnicity/MEGAndroid#28]
ScanQRActivity.java needs to be rewritten using Camera2 or another library in order to get rid of the deprecation warnings.

**Comment by JoshuaLyle** [comment]
So we I've recently moved the min api level from 11 (Android 3.0) to 14 (Android 4.0) because I need 14 to use sliders in the advanced options menu. Our current target version is 23 (Android 6.0).

Camera2 was introduced in 21 (Android 5.0) so we can either keep backwards compatibility with 4.x devices at level 14 as we currently have it and close this issue or move the level to 21 and only support Android 5.x and on.

My 2¢ is that keeping backwards compatibility is more favorable however I have had a few autofocus issues with other new devices that might be related to the old library.

Thoughts?

**Comment by bbusenius** [comment]
If memory serves, we needed to upgrade to a newer version of appcompat in order to enable password toggle and some of the other visual niceties from the comp. This is what I was trying to do in a64cfddb3825fa00a14bb2ad3c37bb20e66448ae so I could knock out some of the easier design related issues. When I last looked at it, the best I could figure was that upgrading the dependencies caused instability and crashing after the QR Scanner was launched (or trying to launch). I opened a bug for this here: #26 .  From what I could tell, this was happening because of the camera and since @hahnicity had called out the deprecated library in his TODO comments in the source, I opened this ticket. 

**My thinking is that if we can get rid of the instability and crashing in #26 and achieve all of the design specs without upgrading, I'm comfortable deferring to your judgment on this.** The only caveat is that since the libraries are very different, the reworking of the camera code represents a sizeable amount of work and part of me thinks it would be nice to get it over with so it's not hanging over us but that's not critical at this juncture.
